appserver/web/web-core/src/main/java/fish/payara/appserver/web/core/CatalinaWebStack.java

```diff
@@ -97,8 +105,20 @@ public void addHost(Host host) {
         engine.addChild(host);
     }
 
+    public Container[] getHosts() {
+        return engine.findChildren();
+    }
+
+    public Host getHost(String name) {
+        return (Host) engine.findChild(name);
+    }
+
     public void start() throws LifecycleException {
-        server.start();
+        if (!hasConnectors) {
+            service.addConnector(new GrizzlyConnector());
+        }
+        tomcat.init();
```

Comment:
Yes, that's what appears to be correct startup sequence for Tomcat. We don't need explicit `init` command, because it is tomcat implementation detail not relevant for WebContainer.